src/main/java/org/opensearch/knn/index/KNNSettings.java

```diff
@@ -390,14 +378,14 @@ public static ByteSizeValue parseknnMemoryCircuitBreakerValue(String sValue, Str
     }
 
     /**
-     * Updates knn.circuit_breaker.triggered setting to true/false
-     * @param flag true/false
+     * Updates boolean setting to true/false
+     * @param value true/false
      */
-    public synchronized void updateCircuitBreakerSettings(boolean flag) {
+    public synchronized void updateBooleanSetting(String settingName, boolean value) {
         ClusterUpdateSettingsRequest clusterUpdateSettingsRequest = new ClusterUpdateSettingsRequest();
-        Settings circuitBreakerSettings = Settings.builder().put(KNNSettings.KNN_CIRCUIT_BREAKER_TRIGGERED, flag).build();
+        Settings circuitBreakerSettings = Settings.builder().put(settingName, value).build();
```

Comment:
what will happen if the setting name is not found? We should handle that case also.

Reply:
With updated log message:
```
»  java.lang.IllegalArgumentException: persistent setting [invalid-setting-lets-see-what-happens], not recognized
»       at org.opensearch.common.settings.AbstractScopedSettings.updateSettings(AbstractScopedSettings.java:863) ~[opensearch-2.5.1-SNAPSHOT.jar:2.5.1-SNAPSHOT]
»       at org.opensearch.common.settings.AbstractScopedSettings.updateDynamicSettings(AbstractScopedSettings.java:810) ~[opensearch-2.5.1-SNAPSHOT.jar:2.5.1-SNAPSHOT]
»       at org.opensearch.action.admin.cluster.settings.SettingsUpdater.updateSettings(SettingsUpdater.java:111) ~[opensearch-2.5.1-SNAPSHOT.jar:2.5.1-SNAPSHOT]
»       at org.opensearch.action.admin.cluster.settings.TransportClusterUpdateSettingsAction$1.execute(TransportClusterUpdateSettingsAction.java:254) ~[opensearch-2.5.1-SNAPSHOT.jar:2.5.1-SNAPSHOT]
»       at org.opensearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:65) ~[opensearch-2.5.1-SNAPSHOT.jar:2.5.1-SNAPSHOT]
»       at org.opensearch.cluster.service.MasterService.executeTasks(MasterService.java:867) ~[opensearch-2.5.1-SNAPSHOT.jar:2.5.1-SNAPSHOT]
»       at org.opensearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:424) ~[opensearch-2.5.1-SNAPSHOT.jar:2.5.1-SNAPSHOT]
»       at org.opensearch.cluster.service.MasterService.runTasks(MasterService.java:295) [opensearch-2.5.1-SNAPSHOT.jar:2.5.1-SNAPSHOT]
»       at org.opensearch.cluster.service.MasterService$Batcher.run(MasterService.java:206) [opensearch-2.5.1-SNAPSHOT.jar:2.5.1-SNAPSHOT]
»       at org.opensearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:190) [opensearch-2.5.1-SNAPSHOT.jar:2.5.1-SNAPSHOT]
»       at org.opensearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:228) [opensearch-2.5.1-SNAPSHOT.jar:2.5.1-SNAPSHOT]
»       at org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:747) [opensearch-2.5.1-SNAPSHOT.jar:2.5.1-SNAPSHOT]
»       at org.opensearch.common.util.concurrent.PrioritizedOpenSearchThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedOpenSearchThreadPoolExecutor.java:282) [opensearch-2.5.1-SNAPSHOT.jar:2.5.1-SNAPSHOT]
»       at org.opensearch.common.util.concurrent.PrioritizedOpenSearchThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedOpenSearchThreadPoolExecutor.java:245) [opensearch-2.5.1-SNAPSHOT.jar:2.5.1-SNAPSHOT]
»       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
»       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
»       at java.lang.Thread.run(Thread.java:834) [?:?]
```